AIFoV/src/main/java/dk/gruppeseks/aifov/AIFoVProvider.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.gruppeseks.aifov;
+
+import dk.gruppeseks.bodtrd.common.data.Entity;
+import dk.gruppeseks.bodtrd.common.data.Path;
+import dk.gruppeseks.bodtrd.common.data.World;
+import dk.gruppeseks.bodtrd.common.data.entityelements.AIData;
+import dk.gruppeseks.bodtrd.common.data.entityelements.Body;
+import dk.gruppeseks.bodtrd.common.data.entityelements.Position;
+import dk.gruppeseks.bodtrd.common.data.util.Vector2;
+import dk.gruppeseks.bodtrd.common.exceptions.NoPathException;
+import dk.gruppeseks.bodtrd.common.services.AISPI;
+import java.util.ArrayList;
+import java.util.List;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Morten
+ */
+@ServiceProvider(service = AISPI.class)
+public class AIFoVProvider implements AISPI
+{
+    private World _world;
+    private final int RAY_COUNT = 100;
+    private final int VISION_DISTANCE = 350;
+    private final double VISION_DEGREES = 90;
+    private final int DUMB_AI_RANGE = 80;
+
+    @Override
+    public Path getPath(Entity entity, Entity target, World world) throws NoPathException
+    {
+        List<Position> path = new ArrayList<>();
+        _world = world;
+        Position targetPos = rayCast(entity, target); // Providing the point on the circle it collided would result in better AI.
+
+        if (targetPos == null)
```

Comment:
If rayCast() returns null it's cause it didnt find a player, and thus the zombie shouldnt move (path should be empty). In this case it could be imagined that the zombie would get a idle path, but currently it doesnt.